akka-stream/src/main/scala/akka/stream/impl/QueueSource.scala

```diff
@@ -53,9 +53,13 @@ import scala.util.control.NonFatal
         if (maxBuffer > 0) buffer = Buffer(maxBuffer, materializer)
         initCallback(callback.invoke)
       }
-      override def postStop(): Unit = stopCallback {
-        case Offer(elem, promise) ⇒ promise.failure(new IllegalStateException("Stream is terminated. SourceQueue is detached"))
-        case _                    ⇒ // ignore
+      override def postStop(): Unit = {
+        val illegalStateException = new IllegalStateException("Stream is terminated. SourceQueue is detached")
```

Comment:
Should it maybe rather be the `AbruptStageTerminationException` which the other stages use in `postStop`?